www/src/components/semesters/SemesterCalendar.vue

```diff
@@ -69,10 +70,19 @@ const teachers = computed<UserPublicData[]>(() => {
     return Object.values(teachersById).map<UserPublicData>(teachers => teachers![0]!);
 });
 
-const selectedSubjects = ref<SubjectData[]>([]);
 const selectedClassrooms = ref<ClassroomData[]>([]);
 const selectedTeachers = ref<UserPublicData[]>([]);
 
+const subjectGroupSelections = ref<Record<string, SubjectData[]>>({});
+
+subjectGroups.forEach(group => {
+    subjectGroupSelections.value[group.title] = [];
+});
```

Comment:
```suggestion
watchEffect(() => {
    subjectGroupSelections.value = subjectGroups.reduce(
        (selections, group) => {
            selections[group.title] = [];
            return selections;
        },
        {} as Record<string, SubjectData[]>,
    );
});
```

The groups likely won't change, but when using props/state to compute new values, it's nice to make sure it will respect the potential changes.